version/no_check.go

```diff
@@ -18,24 +18,10 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-// +build thriftrw.disableVersionCheck
-
 package version
 
-// CheckCompatWithGeneratedCodeAt will panic if the ThriftRW version used to
-// generated code (given by `genCodeVersion`) is not compatible with the
-// current version of ThriftRW.
-// This function is designed to be called during initialization of the
-// generated code.
-//
-// Rationale: It is possible that the old generated code is not compatible with
-// the new version of ThriftRW in subtle ways but still compiles successfully.
-// This function will ensure that the version mismatch is detected and help
-// avoid bugs that could be caused by this discrepancy.
-func CheckCompatWithGeneratedCodeAt(string, string) {
-	// This version does not do anything. It is used only when the
-	// disableVersionCheck is used. Use the following command to build a
-	// binary with this tag.
-	//
-	// 	go build -tags thriftrw.disableVersionCheck
-}
+// CheckCompatWithGeneratedCodeAt is a relic from when we attempted to verify
+// that generated code was installed with the vendored version of thriftrw.
+// It had a troubled past, forcing dependees to regenerate code whenever
```

Comment:
I think the version check makes sense in one direction: disallowing generated code from version N+1 to work with library version N. However, the other direction (which is what this comment calls out) doesn't really make sense: If you generate code at version N, then library versions N+1 (within the same major) should be compatible.

Is the issue here that we were doing the wrong check?

Reply:
We are doing the wrong check, so every `glide up` must be followed by codegen. This is particularly bad because codegen tends to also update mocks, which in turn leak implementation details that would not otherwise constitute a breaking change.

The current check verifies that the code was not generated with a prior release, and at that, appears to have a bug since 1.9.0 does not satisfy 1.9.0.

This is a third-order yak shave. My inclination is to remove the check and introduce the correct check later.

Reply:
Sure I'm ok with that. If it's easy to flip the version check to do the opposite, I'd prefer that, but I'm OK with whatever approach is simplest for now.